zqd/handlers.go

```diff
@@ -98,14 +98,18 @@ func handleSearch(c *Core, w http.ResponseWriter, r *http.Request) {
 	defer srch.Close()
 
 	var out search.Output
+	ctrl := true
+	if r.URL.Query().Get("noctrl") != "" {
+		ctrl = false
+	}
 	format := r.URL.Query().Get("format")
 	switch format {
 	case "zjson", "json":
 		// XXX Should write appropriate ndjson content header.
-		out = search.NewJSONOutput(w, search.DefaultMTU)
+		out = search.NewJSONOutput(w, search.DefaultMTU, ctrl)
 	case "zng":
 		// XXX Should write appropriate zng content header.
-		out = search.NewZngOutput(w)
+		out = search.NewZngOutput(w, ctrl)
```

Comment:
If ctrl == false, we could truthfully set the content-header to specify that pure zng is returned.

Reply:
I think it would be truthful to send that content header even if `ctrl == true`, because the in line control messages are part of the zng spec, which specifies that "Any such application-specific payloads not known by a ZNG data receiver shall be ignored.".

(Yes, this means that it should be ok for the client to get data without even passing this new `noctrl` query param... but i it's still a bit nicer not to have stats and such messages lying around in downloaded zngs)

Reply:
(And re the original comment, I'll file an issue to fix the content-headers in a follow-up.)

Reply:
> I'll file an issue 

#745